test/integration/extensions_test.go

```diff
@@ -0,0 +1,229 @@
+package integration
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+func TestExtensions(t *testing.T) {
+	// Create a temporary directory.
+	tmpdir, err := ioutil.TempDir("", "extensions")
+	if err != nil {
+		t.Fatalf("Could not create tmp dir for extensions: %v", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	// Create extension files.
+	masterOptions := setupExtensions(t, tmpdir)
```

Comment:
you could simplify the setup by having a map of the file structure you want and just iterating over it and creating the files:

var testData = map[string]string {
  "script1.js": script1,
  "script2.js": script2,
  "stylesheet1.css": stylesheet1,
  "stylesheet2.css": stylesheet2,
  "extension1/index.html": index,
  "extension2/index.html": index,
}

Then the setup becomes about 10 lines